engine/swarm/secrets.md

```diff
@@ -0,0 +1,837 @@
+     \_ redis.1.2s6yorvd9zow  redis:alpine  moby  Shutdown       Failed 56 seconds ago      "task: non-zero exit (1)"  
+     \_ redis.1.ulfzrcyaf6pg  redis:alpine  moby  Shutdown       Failed about a minute ago  "task: non-zero exit (1)"  
+     \_ redis.1.wrny5v4xyps6  redis:alpine  moby  Shutdown       Failed 2 minutes ago       "task: non-zero exit (1)"
+    ```
+
+4.  Get the ID of the `redis` service task container using `docker ps` , so that
+    you can use `docker exec` to connect to the container and read the contents
+    of the secret data file, which defaults to being readable by all and has the
+    same name as the name of the secret. The first command below illustrates
+    how to find the container ID, and the second and third commands use shell
+    completion to do this automatically.
+
+    ```bash
+    $ docker ps --filter name=redis -q
+
+    5cb1c2348a59
+
+    $ docker exec $(docker ps --filter name=redis -q) ls -l /run/secrets
+
+    total 4
+    -r--r--r--    1 root     root            17 Dec 13 22:48 my_secret_data
+
+    $ docker exec $(docker ps --filter name=redis -q) cat /run/secrets/my_secret_data
+
+    This is a secret
+    ```
+
+5.  Verify that the secret is **not** available if you commit the container.
+
+    ```bash
+    $ docker commit $(docker ps --filter name=redis -q) committed_redis
+
+    $ docker run --rm -it committed_redis cat /run/secrets/my_secret_data
+
+    cat: can't open '/run/secrets/my_secret_data': No such file or directory
+    ```
+
+6.  Try removing the secret. The removal fails because the `redis` is running
+    and has access to the secret.
+
+    ```bash
+
+    $ docker secret ls
+
+    ID                          NAME                CREATED             UPDATED
+    wwwrxza8sxy025bas86593fqs   my_secret_data      4 hours ago         4 hours ago
+
+
+    $ docker secret rm my_secret_data
+
+    Error response from daemon: rpc error: code = 3 desc = secret 'my_secret_data' is in use by the following service: redis
+    ```
+
+7.  Remove access to the secret from the running `redis` service by updating the
+    service.
+
+    ```bash
+    $ docker service update --secret-rm="my_secret_data" redis
+    ```
+
+8.  Repeat steps 3 and 4 again, verifying that the service no longer has access
+    to the secret. The container ID will be different, because the
+    `service update` command redeploys the service.
+
+    ```bash
+    $ docker exec -it <container_id> cat /run/secrets/my_secret_data
+
+    cat: can't open '/run/secrets/my_secret_data': No such file or directory
+    ```
+
+7.  Stop and remove the service, and remove the secret from Docker.
+
+    ```bash
+    $ docker service rm redis
+
+    $ docker secret rm my_secret_data
+    ```
+
+### Intermediate example: Use secrets with a Nginx service
+
+> **Note**: This example uses a single-Engine swarm for simplicity, and only
```

Comment:
Non-blocking:  Since all of these examples (redis, nginx, and wordpress) presume a single-Engine swarm, should that part of this note be moved to the top of the examples section?  (Similarly for the wordpress example)